File: Source/Rio.Web/src/app/shared/models/market-metrics-dto.ts
import { OfferDto } from './offer/offer-dto';
import { WaterTransferDto } from './water-transfer-dto';

export class MarketMetricsDto {
    MostRecentOfferToBuyPrice: number;
    MostRecentOfferToBuyQuantity: number;
    MostRecentOfferToSellPrice: number;
    MostRecentOfferToSellQuantity: number;
    MostRecentWaterTransfer: WaterTransferDto;
    TotalBuyVolume: number;
    TotalSellVolume: number;

    constructor(obj?: any) {
        Object.assign(this, obj);
    }
}

File: Source/Rio.Models/DataTransferObjects/MarketMetricsDto.cs
using Rio.Models.DataTransferObjects.Offer;
using Rio.Models.DataTransferObjects.WaterTransfer;

namespace Rio.Models.DataTransferObjects
{
    public class MarketMetricsDto
    {
        public int? MostRecentOfferToBuyQuantity { get; set; }
        public decimal? MostRecentOfferToBuyPrice { get; set; }
        public int? MostRecentOfferToSellQuantity { get; set; }
        public decimal? MostRecentOfferToSellPrice { get; set; }
        public WaterTransferDto MostRecentWaterTransfer { get; set; }
        public int TotalBuyVolume { get; set; }
        public int TotalSellVolume { get; set; }
    }
}